src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateCommandRunner.cs

```diff
@@ -278,194 +275,93 @@ bool PackageHasVulnerability(string packageId, NuGetVersion version, IReadOnlyLi
-        if (sourceMapping.IsEnabled)
-        {
-            logger.LogMinimal(Strings.Unsupported_PackageSourceMapping, ConsoleColor.Red);
-            return new List<PackageUpdateResult>();
-        }
-
-        var allProjectPackages = GetAllPackagesReferencedByProject(project);
-        var packagesToUpdate = new List<PackageUpdateResult>();
 
-        foreach (var package in allProjectPackages)
-        {
-            // GetAllPackagesReferencedByProject returns packages with the requested version, so clear it since we want to find the highest version.
-            var packageToCheck = package with { VersionRange = null };
-            var result = await GetSinglePackageToUpdateAsync(packageToCheck, project, versionChooser, settings, logger, suppressWarnings: true, cancellationToken);
-            // GetSinglePackageToUpdateAsync returns null if the package is already using the highest version.
-            // This is not an error when updating all packages in a project.
-            if (result != null)
+            VersionRange upgradeVersion;
+            if (package.VersionRange is not null)
             {
-                packagesToUpdate.Add(result);
+                upgradeVersion = package.VersionRange;
+                if (upgradeVersion == existingVersion)
+                {
+                    logger.LogMinimal(Messages.Warning_AlreadyUsingSameVersion(package.Id, upgradeVersion.OriginalString), ConsoleColor.Yellow);
+                    continue;
+                }
             }
-        }
-
-        return packagesToUpdate;
-    }
-
-    private static List<Package> GetAllPackagesReferencedByProject(PackageSpec project)
-    {
-        var allPackageIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-        var allPackages = new List<Package>();
-
-        foreach (var tfm in project.TargetFrameworks)
-        {
-            foreach (var dependency in tfm.Dependencies)
+            else
             {
-                // Skip auto-referenced packages and packages without explicit names
-                if (!string.IsNullOrEmpty(dependency.Name) && !dependency.AutoReferenced)
+                bool usePrerelease = existingVersion.HasLowerBound && existingVersion.MinVersion.IsPrerelease;
+                var latestVersion = await packageUpdateIO.GetLatestVersionAsync(package.Id, usePrerelease, NullLogger.Instance, cancellationToken);
+                if (latestVersion is null)
                 {
-                    if (allPackageIds.Add(dependency.Name))
-                    {
-                        allPackages.Add(new Package
-                        {
-                            Id = dependency.Name,
-                            VersionRange = dependency.LibraryRange.VersionRange
-                        });
-                    }
+                    logger.LogMinimal(Messages.Error_NoVersionsAvailable(package.Id), ConsoleColor.Red);
```

Comment:
I didn't notice it earlier when I reviewed this code (or most likely I forgot), but wouldn't it be more appropriate to call `logerror` and log warning instead of specifying the color?

Is it because of the old prepend word experience we have in our dotnet commands or something else?

My idea is if that's the reason, then we can maybe implement a logger that actually does use color and pass that to update command. 
That way the update command doesn't add technical debt itself.

Reply:
Honestly, I no longer remember why I implemented it this way.  Doing a quick local test, I feel like yellow and red text gives a better experience than the "error:" at the beginning of the line and all white text. 

However, this does make me remember that a conversation I had with Chet where we should really be following UNIX conventions of writing warnings and errors to stderr instead of stdout.

I'm not sure if we should change the existing CommandOutputLogger, therefore affecting all commands that use it, or if we should create a new logger.

I think we can consider this for a future improvement. I'd like to get this PR in and finish implementing other features before looking to change unrelated nuget commands in the dotnet cli.